Feature

FenZlo opened the pull request:
Мужчины, доброе пожаловать на мою первую ветку

FenZlo committed:
First Commit

Первый коммент

FenZlo committed:
First commit from Fen

*Create index.HTML, style.css and sript.js

FenZlo committed:
Changes with JS

vgneery commented on script.js:
```diff
@@ -1 +1,25 @@
+    let textOfBusiness = document.createElement('div')
+    
+    let elementAddString = document.getElementById('addInput').value; //присваиваю новой переменной значение, которое ввел в поле ввода
+    if (elementAddString.lenth<0) return; //Проверка на то, что поле ввода не пустое
```
свойства lenth нет, тут опечаточка
```suggestion
    if (elementAddString.length<0) return; //Проверка на то, что поле ввода не пустое
```

vgneery replied:
ну и условие кажется не совсем правильным, меньше нуля длинна строки просто не может быть)

vgneery commented on script.js:
```diff
@@ -1 +1,25 @@
+    console.log(elementAddString);
+    
+    textOfBusiness.appendChild(elementAddString); //добавляю значение из поля ввода в newElement, который создал ранее
+    textOfBusiness.appendChild(newElement);
```
кажется мы тут хотим в newElement добавлять ноды, а не наоборот

FenZlo replied:
Почему-то при попытке добавить текст из поля ввода в дивчик textOfBusiness  пишет, что не удалось выполнить appendChild, так как параметр 1 не относится к Node. 

Чет не могу понять, почему он не может туда значение перенести

Код, если что переписал так:
textOfBusiness.appendChild(elementAddString);
newElement.appendChild(textOfBusiness);

И он на первом же добавлении спотыкается

vgneery commented on script.js:
```diff
@@ -1 +1,25 @@
+    textOfBusiness.appendChild(elementAddString); //добавляю значение из поля ввода в newElement, который создал ранее
+    textOfBusiness.appendChild(newElement);
+    
+    let trash = document.createElement('img')
+    trash.src = './to-do-list/trash.png'
+    newElement.appendChild(trash);
```
будет прикольно, если разнесешь генерация разных элементов в разные функции, например

FenZlo replied:
не очень понял, что нужно сделать, если честно

AtnesNess replied:
он имеет ввиду, что ты можешь создать отдельный function, который создаст и вернет элемент. А родительская функция уже вызовет ту функцию для создания, и полученный элемент добавит в твой todo list

FenZlo committed:
Внес изменения в appennd

Значение из elementAddString я добавляю через append, вместо
appendChild и теперь это значение уу меня хотя бы заносится в дивчик
list. Я не очень понял, как это работает. И ещё столкнулся с 2
проблемами:
1. Каждый новый элемент добавляется справа от предыдущего, а не под ним.
   Я думаю, что можно попробовать создать в HTML контейнер ul и туда
   добавлть li, так? Или есть какое-то другое решение? Например, можно
   как-то прописать, чтобы новые элементы вставали друг под другом?
2. Почему-то добавляю картинку через JS бразуер её не видит. Я как-то
   синтаксис нарушаю? Потому что я делал так, как было сказано в
   документации. Очепяток вроде тоже нет.

FenZlo committed:
Исправил неработающие косяки

Теперь все работает исправно. Посмтри сам код и CSS и HTML тоже плиз,
вдруг там тоже есть косячеллы

AtnesNess commented on index.html:
```diff
@@ -1 +1,29 @@
+    <link rel="stylesheet" href="./style.css">
+</head>
+
+<body class="body">
```
на body особо нет смысла классы навешивать, этот тег всегда один на странице

AtnesNess commented on index.html:
```diff
@@ -1 +1,29 @@
+</head>
+
+<body class="body">
+    <div class="title">
```
кроме div существует еще много разных тэгов. https://developer.mozilla.org/en-US/docs/Web/HTML/Element. Старайся его использовать тольку в случае если это обертка без особого семантического смысла

Например здесь можно было бы использовать header https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header

AtnesNess commented on index.html:
```diff
@@ -1 +1,29 @@
+    <div class="title">
+        <h1>TO-DO LIST</h1>
+    </div>
+    <div class="stringForAdd">
```
также старайся называть теги по какой нибудь методологии. Например по БЭМ https://ru.bem.info/methodology/key-concepts/

AtnesNess replied:
Я кажется свою лекцию прикладывал в предыдущем Пул реквесте, оч рекомендую посмотреть

AtnesNess commented on script.js:
```diff
@@ -1 +1,33 @@
+        
+    textOfBusiness.append(elementAddString); //добавляю значение из поля ввода в newElement, который создал ранее
+    newElement.appendChild(textOfBusiness);
+    console.log(newElement);
```
кажется лишнее затесалось

AtnesNess commented on script.js:
```diff
@@ -1 +1,33 @@
+    newElement.appendChild(textOfBusiness);
+    console.log(newElement);
+    
+    let trash = document.createElement('img')
```
чтобы муторно не прописывать всю структуру, а писать в привычной разметке - можно пользоваться template'ами https://www.w3schools.com/tags/tag_template.asp

AtnesNess commented on script.js:
```diff
@@ -1 +1,33 @@
+    
+    toDoList.appendChild(newElement); //Перемещаю div в список дел, который создал заранее в HTML и объявил его в начале JS
+    
+    trash.addEventListener('click', deletText);
```
получается что ты добавляешь обработчик на каждый элемент списка. Что считается не очень хорошим тоном и при оч большом кол-ве элементов начнет тормозить браузер. 

Почитать можно тут: https://learn.javascript.ru/event-delegation

AtnesNess commented on style.css:
```diff
@@ -1 +1,76 @@
+    height: 30px;
+    background-color: rgb(255, 255, 255);
+    width: 400px;
+    display: flex;
```
кстати про флексбоксы, чтоы лучше запомнилось и усвоилось. Вот тебе пару игр
http://flexboxfroggy.com/ 
http://www.flexboxdefense.com/

AtnesNess commented on style.css:
```diff
@@ -1 +1,76 @@
+    display: flex;
+    justify-content: space-between;
+    margin:0 auto;
+    padding: 0 10px 0 10px;
```
значение padding можно писать сокращенно. 
Запись 
padding: 10px означает что все стороны будут иметь одинаковый падинг
padding: 10px 5px; означает что padding внизу и вверху 10px а по бокам 5px
padding: 10px 9px 8px 7px; расставляет все 4 стороны
ну и padding: 10px 9px 8px; установит 10px  сверху, 9px справа, 8px снизу И 9px слева.

AtnesNess commented on script.js:
```diff
@@ -1 +1,33 @@
+    
+    let trash = document.createElement('img')
+    trash.classList.add('trash');
+    trash.src = './trash.png'
```
насчет картинок, png обычно оч тяжелые и рекомендуется использовать svg так как они не растровые, а векторные. И хорошо смотрятся при любом масштабе 

Про растр vs вектор: https://npt.ru/blog/rastr-ili-vektor-rukovodstvo-dlja-chajnikov/
